### PR TITLE
[Core] Hexhedron3D8 - adding methods for metric computations

### PR DESCRIPTION
**Pull request by ddiezrod**
This adds the following methods for hexahedra3d8.
- ComputeSolidAngles
- ComputeDihedralAngles
- MinDihedralAngle
- MaxDihedralAngle
- VolumeToRMSEdgeLength
- ShortestToLongestEdgeQuality

### CODE REVIEW

**Review by pablobecker — DISMISSED** [review]

**Review by loumalouomega — COMMENTED** [review]
This could be more efficient, I will do a minor change, wait

**Review by loumalouomega — DISMISSED** [review]
Now

**Review by loumalouomega — DISMISSED** [review]

File: kratos/geometries/hexahedra_3d_8.h [hunk 1]
```diff
@@ -717,8 +740,90 @@ template<class TPointType> class Hexahedra3D8 : public Geometry<TPointType>
...
+
+    /** 
+     * @brief Calculates the max dihedral angle quality metric.
+     * @details Calculates the max dihedral angle quality metric.
+     * The max dihedral angle is max angle between two faces of the element
+     * in radians
+     * @return [description]
+     */
+    double MaxDihedralAngle() const override {
+        Vector dihedral_angles(24);
```
Comment (loumalouomega) [inline kratos/geometries/hexahedra_3d_8.h:774]: Use an array if the size is fixed

Comment (loumalouomega) [reply kratos/geometries/hexahedra_3d_8.h:774]: Ah, Ok the method works with Vector, sorry

File: kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp [hunk 2]
```diff
@@ -309,16 +309,52 @@ namespace Kratos {
...
-        Vector result_solid_angles = ZeroVector(8);
-        result_solid_angles[0] = Globals::Pi/4.0;
-        result_solid_angles[1] = 3.0*Globals::Pi/4.0;
-        result_solid_angles[2] = 3.0*Globals::Pi/4.0;
-        result_solid_angles[3] = Globals::Pi/4.0;
-        result_solid_angles[4] = 3.0*Globals::Pi/4.0;
-        result_solid_angles[5] = Globals::Pi/4.0;
-        result_solid_angles[6] = Globals::Pi/4.0;
-        result_solid_angles[7] = 3.0*Globals::Pi/4.0;
+        const double pi = Globals::Pi;
```
Comment (loumalouomega) [inline kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp]: constexpr

File: kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp [hunk 3]
```diff
@@ -309,16 +309,52 @@ namespace Kratos {
...
+        };
         KRATOS_CHECK_VECTOR_NEAR(solid_angles, result_solid_angles, TOLERANCE);
     }
+
+    /** Checks the diahedral angles */
+    KRATOS_TEST_CASE_IN_SUITE(Hexahedra3D8DihedralAngles, KratosCoreGeometriesFastSuite) {
+        auto geom = GenerateDeformedCenterLen1Hexahedra3D8();
+        Vector diahedral_angles;
+        geom->ComputeDihedralAngles(diahedral_angles);
+        const double pi = Globals::Pi;
```
Comment (loumalouomega) [inline kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp]: constexpr

**Review by loumalouomega — DISMISSED** [review]
Okay

File: kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp (hunk 2 above)
Comment (ddiezrod) [reply kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp]: done

File: kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp (hunk 3 above)
Comment (ddiezrod) [reply kratos/tests/cpp_tests/geometries/test_hexahedra_3d_8.cpp]: done

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**